enterprise/server/backends/authdb/authdb.go

```diff
@@ -435,27 +446,35 @@ func (d *AuthDB) GetAPIKeyGroupFromAPIKeyID(ctx context.Context, apiKeyID string
 
 func (d *AuthDB) LookupUserFromSubID(ctx context.Context, subID string) (*tables.User, error) {
 	user := &tables.User{}
-	err := d.h.TransactionWithOptions(ctx, db.Opts().WithStaleReads(), func(tx interfaces.DB) error {
-		rq := tx.NewQuery(ctx, "authdb_lookup_user_from_subid").Raw(
-			`SELECT * FROM "Users" WHERE sub_id = ? ORDER BY user_id ASC`, subID)
-		if err := rq.Take(user); err != nil {
-			return err
-		}
-		rq = tx.NewQuery(ctx, "authdb_lookup_user_groups").Raw(`
+	err := d.h.NewQueryWithOpts(
+		ctx,
+		"authdb_lookup_user_from_subid",
+		db.Opts().WithStaleReads(),
+	).Raw(
+		`SELECT * FROM "Users" WHERE sub_id = ? ORDER BY user_id ASC`,
+		subID,
+	).Take(user)
+
+	if err != nil {
+		return nil, err
+	}
+
+	rq := d.h.NewQueryWithOpts(
+		ctx,
+		"authdb_lookup_user_groups",
+		db.Opts().WithStaleReads(),
+	).Raw(`
 			SELECT g.*, ug.role
 			FROM "Groups" AS g, "UserGroups" AS ug
 			WHERE g.group_id = ug.group_group_id
 			AND ug.membership_status = ?
 			AND ug.user_user_id = ?
-			`, int32(grpb.GroupMembershipStatus_MEMBER), user.UserID,
-		)
-		gs, err := db.ScanAll(rq, &tables.GroupRole{})
-		if err != nil {
-			return err
-		}
-		user.Groups = gs
-		return nil
-	})
+		`,
+		int32(grpb.GroupMembershipStatus_MEMBER),
+		user.UserID,
+	)
+
+	user.Groups, err = db.ScanAll(rq, &tables.GroupRole{})
```

Comment:
It seems to me that queries like this, with  two select statements run, the second depending on the first, should be using REPEATABLE READ isolation level to avoid possible race conditions.

What do you think?

Reply:
Oh interesting, I wasn't aware of REPEATABLE READ. You're definitely right; we could get inconsistency that doesn't currently exist if we don't keep this in a transaction. Not that I'm looking at doing this right now, but would it make sense if we moved some of these that don't have meaningful application logic in the middle into `JOIN`s?

Reply:
yeah I think if we removed the two-dependent-select statements from this CL, then sent another CL to make those JOINs (or leave them in a REPEATABLE READ txn) we'd be solid

Reply:
Sounds good!